hollow/src/main/java/com/netflix/hollow/api/producer/validation/DuplicateDataDetectionValidator.java

```diff
@@ -153,4 +159,9 @@ private void handleEndValidation(SingleValidationStatusBuilder statusBuilder, St
 				+ "is not Object. This validation cannot be done.";
 	private static final String FIELD_PATH_NAME = "FieldPaths";
 	private static final String DATA_TYPE_NAME = "Typename";
+
+	@Override
+	public String getName() {
+		return NAME+"_"+dataTypeName;
```

Comment:
I wonder if instead of needing `Nameable` if what we really want is something like a type-specific validator:

```java
interface TypeSpecificValidator extends Validator {
  String getTypeName();
}
```

Then in the `HollowProducer::validate` construct the name by combining the validator class name with the type name, as you're doing here.

Reply:
What you suggest is one way to achieve this visibility. But at this point, leaving it to validators whether to even have a name is fine. In next iteration we can focus on how we would like HollowProducer and validator relation to change.